### PR TITLE
RLP-512 Settlement Light

### PR DESCRIPTION
**Pull request by debrayperalta**
This PR aims to include in the LC the settlement channel feature, as well as providing callbacks for the developer to show when a channel has successfuly settled.

This PR includes the next changes:

New Features 📚
- A new message type for the settlement
- The settlement successful callback
- A manager for the settlement message
- A boolean in the channel indicating if it was settled (isSettled)
- A new channel state (SETTLED)

Improvements and bug fixes 🐞

- A non setted callback that is triggered will return `undefined` (noop)
- Fixes for certain functions (default values when passing undefined)

Tooling 🛠️ Code format 💅 and tests 🧪

- Changed and fixed the tokens test to use real addresses instead of mocked values (remove some flakiness)
- Added babbel/paser to ESLint so optional chaining is recognized as valid
- Updated Babel, Jest and Babel Plugins to latest version
- Formatted gulpfile
- Babel now has sourcemaps
- Gulp task for watching changes (See Below)
- Added support for [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining)

Gulp Watch Task 🥤

- Now we can use the minified/babelified files for development
- The watch task (Added in gulp) will rerun babel on changes when changing the SDK
- Now we can test changes in development with the code used in production
- Sourcemaps were added, so the debugging experience is much easier

Sorry for the big diff! But the package-lock.json is 95% of the diff itself

### CODE REVIEW

**Comment by marcosmartinez7** [comment]
@JONAF2103 @mauricioirace it would be nice a review from you guys

**Comment by marcosmartinez7** [comment]
This is the acceptance criteria to merge this PR after code review (and also https://github.com/rsksmart/lumino/pull/117)

- Settlement works without any sent or received payment
- Settlement works with payments (both sent and received)
- Settlement from lc is not performed if the other node did it (@JONAF2103 wich is the current behaviour for full nodes? it sends the tx anyway?)

File: src/utils/callbacks.js [hunk 1]
```diff
@@ -63,7 +65,7 @@ const Callbacks = () => {
    */
   const trigger = (name, ...args) => {
     if (callbacks[name]) return callbacks[name](...args);
-    return () => {};
+    return () => undefined;
```
Comment (JONAF2103) [inline src/utils/callbacks.js:68]: Is not better to throw an error here? if the callbackname is wrong you don't want this to keep working right?

**Review by JONAF2103 — APPROVED** [review]
LGTM in general i just leave a comment there

File: src/utils/callbacks.js (hunk 1 above)
Comment (debrayperalta) [reply src/utils/callbacks.js:68]: The thing is that checking throwing and error is a little bit aggressive imo.
A noop is a little bit more subtle in that regard, at most returning a message saying that the specified callback is not supported and will never be triggered is a better way of managing this.

What do you think?

**Comment by JONAF2103** [comment]
This is the acceptance criteria to merge this PR after code review (and also rsksmart/lumino#117)

    Settlement works without any sent or received payment (tested and it works)
    Settlement works with payments (both sent and received) (tested and it works)
    Settlement from lc is not performed if the other node did it (@JONAF2103 wich is the current behaviour for full nodes? it sends the tx anyway?) (normal nodes send both the settlement, normally one of those fail since it is already executed by the other, but the error is handled) I tested this and seems to be working

**Review by marcosmartinez7 — APPROVED** [review]

**Comment by marcosmartinez7** [comment]
@debrayperalta @JONAF2103 this lgtm too, let me know when is ok for you to merge

**Comment by debrayperalta** [comment]
Tested cases 🧪

- LC settless with no payments
- LC settles with payments
- LC tries to settle while partner is up

In all cases the LC handled gracefully the situations and changed the state of the channel, in the cases where payments were present, the balances were correct.

I believe that we should merge this in LC

**Pull request merged**